queries/snakemake/highlights.scm

```diff
@@ -0,0 +1,63 @@
+; inherits: python
+
+;; Compound directives
+["rule" "checkpoint" "module"] @keyword
+
+;; Simple directives
+(directive name: _ @keyword)
```

Comment:
```suggestion
(directive name: _ @field)

(rule_definition name: (identifier) @type)
```

Reply:
Agreed on rule names.

For `@keyword` vs `@field`, see my arguments here: https://github.com/nvim-treesitter/nvim-treesitter/pull/4302#discussion_r1103862281

Reply:
I still don't think it should be keyword, what about attribute, property, label, type?

Reply:
Okay, I think one of these sounds good for the "subordinate" directives (eg. `input`, `output`, which belong to `rule`). Ill think about which one of your suggestions to use. I'm leaning towards `attribute` or `label` currently.

For the "freestanding" directives (eg. `configfile`), which appear at the top level (same level as `rule`), `keyword` still feels appropriate to me. Would you agree to a separation of these?

Reply:
Updated queries to cut down on keyword use: https://github.com/nvim-treesitter/nvim-treesitter/pull/4302/commits/3b4275af2f11875b75de6432195d150acdbe48e7

Please re-open if you disagree.